docs/modules/pipelines/pages/configuring-jobs.adoc

```diff
@@ -169,3 +174,39 @@ cluster change depending on these two settings.
 |suspend job
 |===
 
+== Configuring a User Code Namespace
+
+Normal Jet jobs can be configured to use a xref:clusters:user-code-namespaces.adoc[User Code Namespace] (UCN), which provides them
+access to custom resources associated with that UCN. {ucn} must be enabled and a namespace configured with the desired resources
+created before it can be used in jobs. Any data structures used in these jobs should also have their UCN configured to ensure
+operations for that data structure also have access to the necessary custom resources. Light jobs do not support UCN.
+
+{ucn} are managed separately to the Jet system, so these resources persist between individual jobs. This support allows for jobs to
+be un-deployed and re-deployed while still being able to use the same custom resources within pipelines.
+
+WARNING: {ucn} can be dynamically updated at runtime, but it is recommended that namespaces associated with Jet jobs are not changed after
+initial creation. Doing so may cause job execution inconsistency between member nodes if jobs restart during the namespace update.
```

Comment:
> are not changed after initial creation

this should be clarified. The concern raised in this warning is valid. However, there are safe ways of updating the UCN:

1. update UCN when there are no restarts expected (member termination, autoscaling etc)
2. suspend job, update UCN, resume job
3. https://docs.hazelcast.com/hazelcast/5.5/pipelines/job-update (the heaviest one)

Reply:
I've updated this warning to point towards (2) and also updated (3) to mention UCN updates: https://github.com/hazelcast/hz-docs/pull/1383/commits/70431358252c50d178a69ee6bd062a08a0ea0a8e